packages/date-picker/src/DatePicker/components/Calendar/Calendar.tsx

```diff
@@ -51,50 +46,52 @@ export const Calendar: React.VFC<CalendarProps> = ({
   attributes,
   classNameOverride,
   value,
-  initialMonth,
-  firstDayOfWeek,
+  defaultMonth,
+  weekStartsOn,
   disabledDays,
   onDayChange,
-  range,
   selectedRange,
   modifiers,
-  onKeyDown,
+  mode,
 }) => {
   const calendarRef = useRef<HTMLDivElement>(null)
 
-  const getNavbar = ({ ...navbarProps }: CalendarNavProps) => (
-    <CalendarNav {...navbarProps} />
-  )
-
   // Initial focus when opening the calendar
   useEffect(() => {
     if (!calendarRef.current) return
 
     if (value || selectedRange?.from) {
       const selectedDay = calendarRef.current.getElementsByClassName(
-        "DayPicker-Day--selected"
+        calendarStyles.selected
       )[0] as HTMLElement
-
       selectedDay?.focus()
+      return
     } else {
       const today = calendarRef.current.getElementsByClassName(
-        "DayPicker-Day--today"
+        calendarStyles.today
       )[0] as HTMLElement
-
       today?.focus()
+      return
     }
   }, [])
 
-  const getInitialMonth = () => {
+  const getdefaultMonth = () => {
     if (selectedRange?.from) {
       return selectedRange.from
     } else if (value) {
       return value
     } else {
-      return initialMonth
+      return defaultMonth
     }
   }
 
+  const IconRight = (props: StyledComponent) => (
+    <Icon icon={arrowRight} role="presentation" {...props} />
+  )
+  const IconLeft = (props: StyledComponent) => (
+    <Icon icon={arrowLeft} role="presentation" {...props} />
+  )
```

Comment:
You're not wrong, but the props provided by `StyledComponent` (className, style, children) actually aren't compatible with `Icon`, so we shouldn't be trying to spread them in.

```suggestion
  const IconRight: React.VFC = () => <Icon icon={arrowRight} role="presentation" />
  const IconLeft: React.VFC = () => <Icon icon={arrowLeft} role="presentation" />
```

Alternatively, we could destructure the props and pass them in in a compatible way (eg. `classNameOverride={className}`), but since we aren't even adding extra styles to these icons (unless you did want to?), it doesn't really seem necessary.